src/main/java/dev/corestone/lotrings/abilities/FortuneAbility.java

```diff
@@ -0,0 +1,50 @@
+package dev.corestone.lotrings.abilities;
+
+import dev.corestone.lotrings.LOTRings;
+import dev.corestone.lotrings.Ring;
+import dev.corestone.lotrings.RingState;
+import org.bukkit.Bukkit;
+import org.bukkit.Material;
+import org.bukkit.enchantments.Enchantment;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.block.BlockBreakEvent;
+import org.bukkit.inventory.ItemStack;
+
+public class FortuneAbility extends AbilitySuper {
+
+    private float fortuneLevel;
+
+    public FortuneAbility(LOTRings plugin, Ring ring, String abilityName) {
+        super(plugin, ring, abilityName);
+        try {
+            this.fortuneLevel = plugin.getAbilityDataManager().getAbilityFloatData(abilityName, "fortune-level");
+        } catch (Exception e) {
+            sendLoadError();
+        }
+    }
+
+    @Override
+    public void switchState(RingState ringState) {
+        super.switchState(ringState);
+        if (ringState != RingState.HELD) {
+            HandlerList.unregisterAll(this);
+        } else if (ringState == RingState.HELD) {
+            Bukkit.getPluginManager().registerEvents(this, plugin);
+        }
+    }
+
+    @EventHandler
+    public void onBlockBreak(BlockBreakEvent event) {
+        Player player = event.getPlayer();
+        if (!abilityCanBeUsed(player.getUniqueId())) return;
+
+        ItemStack tool = new ItemStack(Material.DIAMOND_PICKAXE);
+
+        tool.addUnsafeEnchantment(Enchantment.LOOT_BONUS_BLOCKS, (int) fortuneLevel);
```

Comment:
I guess this approach works. Make sure it does not bypass MapProtect please. I'm worried it will break the block anyways from the plugin even when build is denied. If this is the case, set the event priority to the lowest and check if the event was already canceled (research event priorities, MapProtect needs to cancel the event at a higher priority so you can check if the event is already canceled).